src/v/resource_mgmt/tests/cpu_profiler_test.cc

```diff
@@ -58,67 +80,75 @@ SEASTAR_THREAD_TEST_CASE(test_cpu_profiler) {
     BOOST_TEST(results.samples.size() >= 1);
 }
 
-SEASTAR_TEST_CASE(test_cpu_profiler_enable_override) {
+SEASTAR_THREAD_TEST_CASE(test_cpu_profiler_enable_override) {
     // Ensure that overrides to the profiler will enable it and collect samples
     // for the specified period of time.
 
-    ss::sharded<resources::cpu_profiler> cp;
-    co_await cp.start(config::mock_binding(false), config::mock_binding(2ms));
-    co_await cp.invoke_on_all(&resources::cpu_profiler::start);
+    cp_holder holder(false, 2ms);
+    ss::sharded<resources::cpu_profiler>& cp = *holder.cp;
 
     auto wait_ms = 256ms + 10ms;
-    auto [results] = co_await ss::coroutine::all(
-      [&]() {
-          return cp.local().collect_results_for_period(wait_ms, std::nullopt);
-      },
-      [&]() {
-          return ss::smp::invoke_on_all([&]() { return busy_loop(wait_ms); });
-      });
+    auto [results] = ss::when_all_succeed(
+                       [&]() {
+                           return cp.local().collect_results_for_period(
+                             wait_ms, std::nullopt);
+                       },
+                       [&]() {
+                           return ss::smp::invoke_on_all(
+                             [&]() { return busy_loop(wait_ms); });
+                       })
+                       .get();
 
     for (auto& shard_results : results) {
         BOOST_TEST(shard_results.samples.size() >= 1);
     }
 
     // CPU profiler should be disabled so if we wait some time and collect the
     // samples again there should be no new samples.
-    co_await busy_loop(wait_ms);
+    busy_loop(wait_ms).get();
 
     auto new_results = cp.local().shard_results();
-    auto old_results = results[ss::this_shard_id()];
+    auto old_it = std::ranges::find_if(
+      results, [](auto& r) { return r.shard == ss::this_shard_id(); });
```

Comment:
`results` is not in general ordered by shard! this can cause spurious failures

Reply:
(I ended up making it deterministic as this was causing failures in another test and it is cheap to do so)